src/c_api.cpp

```diff
@@ -1034,6 +1034,35 @@ int LGBM_DatasetCreateFromMat(const void* data,
 }
 
 
+static inline std::vector<int32_t> CreateSampleIndices(const Config& config, int32_t total_nrow) {
+  Random rand(config.data_random_seed);
+  int sample_cnt = static_cast<int>(total_nrow < config.bin_construct_sample_cnt ? total_nrow : config.bin_construct_sample_cnt);
+  return rand.Sample(total_nrow, sample_cnt);
+}
+
+
+int LGBM_SampleIndices(int32_t total_nrow,
+                const char* parameters,
+                void* out) {
+  // This API is to keep python binding's behavior the same with C++ implementation.
+  // Sample count, random seed etc. should be provided in parameters.
+  API_BEGIN();
+  if (out == nullptr) {
+    Log::Fatal("sample indices output is nullptr");
+  }
+  auto param = Config::Str2Map(parameters);
+  Config config;
+  config.Set(param);
+
+  auto sample_indices = CreateSampleIndices(config, total_nrow);
+
+  static_assert (sizeof(int) == 4, "int size is not 4");
```

Comment:
What's the purpose of this assert statement?

Reply:
@cyfdecyf

Reply:
This is useless now and removed.

`CreateSampleIndicies` returns `std::vector<int>` at first. It's now returning `std::vector<int32_t>` by implicit type convertion from `Random.Sample`'s return value.